### ISSUE TITLE
1.45.1 breaks *mingw* cross-build config as `UDP_SEGMENT` missing `udp.h` not found

### ISSUE THREAD

**Issue by BrianInglis** [nghttp2/nghttp2#1619]
When you have some time, could you please have a look at why the *mingw* cross-build breaks even though the "native" build works under similar conditions of non-existence and non-definition?
*Cygwin* *mingw* cross-build configs of 1.45.1 break as both `/usr/.../include/netinet/udp.h` and `/usr/.../include/linux/udp.h` not found so `UDP_SEGMENT` is missing. Didn't try 1.45 as the new release came out before I noticed it.
```
checking for timegm... no
checking for timerfd_create... no
checking whether initgroups is declared... no
checking whether UDP_SEGMENT is declared... no
checking whether UDP_SEGMENT is defined as 103 in linux/udp.h... configure: error: in `$HOME/src/cygwin/nghttp2/mingw64-x86_64-nghttp2/mingw64-x86_64-nghttp2-1.45.1-1.noarch/build':
configure: error: cannot run test program while cross compiling
See `config.log' for more details
*** ERROR: configure failed
```
whereas *Cygwin* "native" build handles it, as `netinet/udp.h` exists, but is from *BSD*, and does not define `UDP_SEGMENT`, and `linux/udp.h` does not exist:
```
checking for timegm... yes
checking for timerfd_create... yes
checking whether initgroups is declared... yes
checking whether UDP_SEGMENT is declared... no
checking whether UDP_SEGMENT is defined as 103 in linux/udp.h... no
checking whether C compiler accepts -fvisibility=hidden... yes
configure: updating cache config.cache
checking that generated files are newer than configure... done
configure: creating ./config.status
```
so any cross-build should behave the same as the "native" build where the files do not exist or the symbols are not defined! 

Attached both cross-arch `build/config.log` and cygport process logs, lightly sanitized; `build/config.cache` was 0:

[mingw64-x86_64-nghttp2-build-config.log](https://github.com/nghttp2/nghttp2/files/7230440/mingw64-x86_64-nghttp2-build-config.log)
[mingw64-x86_64-nghttp2-cygport-noarch-all-check.log](https://github.com/nghttp2/nghttp2/files/7230439/mingw64-x86_64-nghttp2-cygport-noarch-all-check.log)

[mingw64-i686-nghttp2-build-config.log](https://github.com/nghttp2/nghttp2/files/7230442/mingw64-i686-nghttp2-build-config.log)
[mingw64-i686-nghttp2-cygport-noarch-all-check.log](https://github.com/nghttp2/nghttp2/files/7230441/mingw64-i686-nghttp2-cygport-noarch-all-check.log)

**Comment by tatsuhiro-t** [comment]
The offending part of configure.ac is removed via commit 407df2822e4b989c53c04b4f7902d2a9210ad506
The check was added specifically for debian 10, but now debian 11 is available, and it would be better to disable advanced features for older systems.

**Comment by BrianInglis** [comment]
Arigatou gozaimasu Tatsuhiro-san! That was quick and greatly appreciated. 
I just found out that I could get your commit as a patch from *Github* by appending `.patch` to *commit id*, and my build system will download the patch of your commit from *Github* then apply it: [Remove check for UDP_SEGMENT patch](https://github.com/nghttp2/nghttp2/commit/407df2822e4b989c53c04b4f7902d2a9210ad506.patch). 
I added that patch URI and both arch build perfectly under Cygwin and Mingw and test perfectly under Cygwin. 
That allows me to distribute and announce those packages, and test the latest release of curl/libcurl with them, then distribute and announce a similar set of packages for curl and libcurl for both arch under both Cygwin and Mingw. 
Thank you very much for all your help. Cheers!
